services/gateway/shared/ready.js

```diff
@@ -34,13 +42,31 @@ const getReady = async broker => {
                 return service;
             },
         );
-        return Promise.resolve({ services: Object.assign({}, ...services) });
+        let allServices;
+        if (config.includeCoreReadiness) {
+            allServices = Object.assign(...services, currentStatus);
+        } else {
+            allServices = Object.assign(...services);
+        }
+
+        const servicesStatus = !Object.keys(allServices).some(value => !allServices[value]);
+        if (servicesStatus) return Promise.resolve({ services: allServices });
+        return Promise.reject(new MoleculerError('503 Not available', 503, 'ERR_SOMETHING'));
```

Comment:
Kindly consider updating to:
```suggestion
        return Promise.reject(new MoleculerError('Core Service Unavailable', 503, 'CORE_SERVICES_NOT_READY', currentStatus));
```